boosting/data_aliases.cc

```diff
@@ -99,7 +104,9 @@ aliases(const ML::Training_Data & dataset, const Feature & predicted)
             auto fv1 = dataset.get(std::get<0>(ex1));
             auto fv2 = dataset.get(std::get<0>(ex2));
 
-            int res = fv1->compare(*fv2, predicted);
+            set<Feature> to_ignore2(to_ignore);
```

Comment:
You can't create a set in the middle of the comparison; it will run horribly slow on any large sized dataset with a significant number of ignored features.  Instead, create a feature_set override that takes both a set and a feature, or alternatively hoist this set out of the lambda so it's only calculated once.